pkg/cli/create/build.go

```diff
@@ -0,0 +1,223 @@
+	cmd.Flags().StringVar(&o.DockerfilePath, "dockerfile-path", o.DockerfilePath, "The relative path within the repository context that the Dockerfile is located at.")
+	cmd.Flags().StringVar(&o.FromImage, "from-image", o.FromImage, "A container image pull spec to use as the basis for the image build.")
+	cmd.Flags().StringVar(&o.ToImage, "to-image", o.ToImage, "A location to push the output image to.")
+	cmd.Flags().StringVar(&o.ToImageStreamTag, "to-image-stream", o.ToImageStreamTag, "An image stream tag to push the output image to. Accepts [NAMESPACE/]STREAM:TAG")
+	cmd.Flags().StringVar(&o.ImageOptimizationPolicy, "image-optimization-policy", o.ImageOptimizationPolicy, "Controls whether individual layers are created: SkipLayers, SkipLayersAndWarn, or None.")
+	cmd.Flags().IntVar(&o.BuildLoglevel, "build-loglevel", o.BuildLoglevel, "Set the log level for builds (0-10, 0 default).")
+	cmd.Flags().StringArrayVar(&o.Env, "env", o.Env, "Add enviroment variables to the build strategy.")
+
+	o.CreateSubcommandOptions.PrintFlags.AddFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+
+	return cmd
+}
+
+func (o *CreateBuildOptions) Complete(cmd *cobra.Command, f genericclioptions.RESTClientGetter, args []string) error {
+	clientConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	o.Client, err = buildv1client.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+
+	return o.CreateSubcommandOptions.Complete(f, cmd, args)
+}
+
+func defaultDockerStrategy(build *buildv1.Build) *buildv1.Build {
+	if build.Spec.CommonSpec.Strategy.DockerStrategy == nil {
+		build.Spec.CommonSpec.Strategy.DockerStrategy = &buildv1.DockerBuildStrategy{}
+	}
+	return build
+}
+
+func stringToImageStreamTagRef(s string) (*corev1.ObjectReference, error) {
+	ref, err := reference.Parse(s)
+	if err != nil {
+		return nil, err
+	}
+	if len(ref.Tag) == 0 || len(ref.ID) > 0 || len(ref.Name) == 0 || len(ref.Registry) > 0 {
+		return nil, fmt.Errorf("must specify [NAMESPACE/]STREAM:TAG")
+	}
+	return &corev1.ObjectReference{
+		Kind:      "ImageStreamTag",
+		Name:      ref.Name + ":" + ref.Tag,
+		Namespace: ref.Namespace,
+	}, nil
+}
+
+func (o *CreateBuildOptions) Run() error {
+	build := &buildv1.Build{
+		// this is ok because we know exactly how we want to be serialized
+		TypeMeta:   metav1.TypeMeta{APIVersion: buildv1.SchemeGroupVersion.String(), Kind: "Build"},
+		ObjectMeta: metav1.ObjectMeta{Name: o.CreateSubcommandOptions.Name},
+		Spec: buildv1.BuildSpec{
+			CommonSpec: buildv1.CommonSpec{
+				Source: buildv1.BuildSource{},
+			},
+		},
+	}
+	if len(o.Strategy) > 0 {
+		switch build.Spec.Strategy.Type {
+		case buildv1.DockerBuildStrategyType:
+			build.Spec.Strategy.DockerStrategy = &buildv1.DockerBuildStrategy{}
+		case buildv1.SourceBuildStrategyType:
+			build.Spec.Strategy.SourceStrategy = &buildv1.SourceBuildStrategy{}
+		case buildv1.CustomBuildStrategyType:
+			build.Spec.Strategy.CustomStrategy = &buildv1.CustomBuildStrategy{}
+		}
```

Comment:
Add `default` which either fails, or screams it defaulted strategy, w/o it we're missing clean information for user what's wrong.

Reply:
Do we have any examples of this in other create commands?  Places where we've added options like volume source where if the client doesn't recognize it it either accepts or errors?

Reply:
We often default a lot in create commands, yours has more than any other create command, but that's reasonable in this particular case. Yet a proper validation here makes sense. Not sure how our apiserver will react to missing strategy type, I can't remember but I bet it will complain. That's why it's better to be more verbose sooner than later.